main.go

```diff
@@ -449,8 +449,8 @@ func linkForType(t *types.Type, c generatorConfig, typePkgMap map[*types.Type]*a
 
 // tryDereference returns the underlying type when t is a pointer, map, or slice.
 func tryDereference(t *types.Type) *types.Type {
-	if t.Elem != nil {
-		return t.Elem
+	for t.Elem != nil {
```

Comment:
it sounds like you mix recursion and for loops here. 

pick one:

1. in a for loop, unpeel t until its t.Elem is nil and return t
2. call func itself recursively if t.Elem is not nil, and return t when t.Elem is nil.

right now you have both combined.

Reply:
Correct!

I will go ahead with the for loop approach to keep it simple and inexpensive.